### PR TITLE
Latest version of Semsimian dependency and search functionality

### PR DESCRIPTION
**Pull request by hrshdhgd**
- [x] Bumped `semsimian` version in toml + lock files.
- [x] Added code for `associations_subject_search()`
- [ ] Test optimization.

### CODE REVIEW

**Comment by cmungall** [comment]
this passes - is there remaining work before this goes draft->merge?

**Review by cmungall — APPROVED** [review]

**Comment by hrshdhgd** [comment]
I wanted to keep this branch to iterate over progressing versions of `semsimian`. The latest release candidate is based off this branch. I intend to merge this once we have a version of `semsimian` that has an acceptable ru time for `termset_pairwise_similarity`

**Comment by cmungall** [comment]
I assume this work has been incorporated in another branch, reopen if necessary

**Pull request closed without merging**
